src/components/sidebar/portaled-sidebar.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC, useState, useEffect } from "react"
+import { Portal } from "react-portal"
+import { PortalSidebox } from "./styled"
+
+const ESCAPE_KEY = 27
+
+export const PortalSidebar: FC<PortalSidebarProps> = ({
+  isOpen = true,
+  children,
+  closeOnEsc = false,
+  onClose = () => {},
+  right = false,
+}: PortalSidebarProps) => {
+  const [isOpenState, setOpen] = useState(isOpen)
+
+  useEffect(() => {
+    setOpen(isOpen)
+    document.addEventListener("keydown", (event: KeyboardEvent) => {
+      if (event.keyCode === ESCAPE_KEY && closeOnEsc) {
+        setOpen(false)
+        onClose()
+      }
+    })
+    return () => {
+      document.removeEventListener("keydown", (event: KeyboardEvent) => {
+        if (event.keyCode === ESCAPE_KEY && closeOnEsc) {
+          setOpen(false)
+          onClose()
+        }
```

Comment:
i think that will not remove event listener properly, you think to pass the exact same function

Reply:
fixed